### PR TITLE
Horizon

### PR DESCRIPTION
**Pull request by jhseceng**
Initial templates and documentation for AWS Control Tower + CSPM (Horizon)

### CODE REVIEW

File: Control-Tower-For-Horizon/documentation/implementation-guide.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+   Fill in the missing parameter values 
+
+   Parameters that are fixed and should only be modifed if advised by CrowdStrike:
+   
+   | Parameter Name | Parameter Description | 
+   | :---         |     :---      | 
+   | CSAccountNumber   | The CrowdStrike account number supplied in the template     |
+   | CSAssumingRoleName   | The name of the Role used by CrowdStrike to assume the role in the account. `CrowdStrikeCSPMConnector`   |
+   | CrowdStrikeCSPMRoleTemplateURL   | The URL for the Template the describes the IAM role to be applied for Horizon access    | YES    |
+   | CrowdStrikeCSPMStackName | The name of the Cloudformation stack applied in the new account |
```
Comment (shawndwells) [inline Control-Tower-For-Horizon/documentation/implementation-guide.md:100]:
```suggestion
   | Parameter Name | Parameter Description | 
   | :-----------------|:-----------------------|
   | CSAccountNumber | The CrowdStrike account number supplied in the template |
   | CSAssumingRoleName | The name of the Role used by CrowdStrike to assume the role in the account. `CrowdStrikeCSPMConnector` |
   | CrowdStrikeCSPMRoleTemplateURL | The URL for the Template the describes the IAM role to be applied for Horizon access | YES |
   | CrowdStrikeCSPMStackName | The name of the Cloudformation stack applied in the new account |
```

File: Control-Tower-For-Horizon/documentation/implementation-guide.md [hunk 2]
```diff
@@ -0,0 +1,191 @@
+# Implementation Guide
```
Comment (shawndwells) [inline Control-Tower-For-Horizon/documentation/implementation-guide.md:1]: There is a LOT in this README. Consider making a table of contents. Checkout https://ecotrust-canada.github.io/markdown-toc/ to help with that.

File: Control-Tower-For-Horizon/lambda/push_horizon_stackset_lambda.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            "sourceIPAddress": "AWS Internal",
+            "userAgent": "AWS Internal",
+            "eventID": "0000000-0000-0000-1111-123456789012",
+            "readOnly": False,
+            "eventType": "AwsServiceEvent",
+            "serviceEventDetails": {
+                "createManagedAccountStatus": {
+                    "organizationalUnit": {
+                        "organizationalUnitName": "Custom",
+                        "organizationalUnitId": "ou-XXXX-l3zc8b3h"
```
Comment (shawndwells) [inline Control-Tower-For-Horizon/lambda/push_horizon_stackset_lambda.py:101]: Is that meant to be hard coded?

File: Control-Tower-For-Horizon/lambda/push_horizon_stackset_lambda.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+                    },
+                    "account": {
+                        "accountName": "LifeCycle1",
+                        "accountId": "XXXXXXXXXXXX"
```
Comment (shawndwells) [inline Control-Tower-For-Horizon/lambda/push_horizon_stackset_lambda.py:106]: Should this be a parameter?

File: Control-Tower-For-Horizon/lambda/push_horizon_stackset_lambda.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+                    },
+                    "state": "SUCCEEDED",
+                    "message": "AWS Control Tower successfully created a managed account.",
+                    "requestedTimestamp": "2019-11-15T11:45:18+0000",
+                    "completedTimestamp": "2019-11-16T12:09:32+0000"}
```
Comment (shawndwells) [inline Control-Tower-For-Horizon/lambda/push_horizon_stackset_lambda.py:111]: Should these be dynamically generated?

File: Control-Tower-For-Horizon/lambda/register_new_horizon_account_v2.py [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+logger.setLevel(logging.INFO)
+
+# CONSTANTS
+SUCCESS = "SUCCESS"
+FAILED = "FAILED"
+
+
+CSPM_ROLE_TEMPLATE_URL = 'https://cs-prod-cloudconnect-templates.s3-us-west-1.amazonaws.com' \
+                         '/aws_cspm_cloudformation_v1.1.json'
+STACK_SET_NAME = "CrowdStrike-CSPM-Integration"
```
Comment (shawndwells) [inline Control-Tower-For-Horizon/lambda/register_new_horizon_account_v2.py:18]: Can this be parameterized to ensure the templates are pulled from within a region? Eg workloads running in GovCloud may not want to connect to other regions.

File: Control-Tower-For-Horizon/s3bucketfiles/cspm-role-template.yaml [hunk 7]
```diff
@@ -0,0 +1,404 @@
+{
+  "AWSTemplateFormatVersion": "2010-09-09",
+  "Description": "Setup script to enable CrowdStrike Falcon CSPM.",
```
Comment (shawndwells) [inline Control-Tower-For-Horizon/s3bucketfiles/cspm-role-template.yaml:3]: Can the product name be used here?

File: Control-Tower-For-Horizon/s3bucketfiles/horizon_new_account_v4.yaml [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+    Type: String
+    NoEcho: 'true'
+
+  CSAccountNumber:
+    AllowedPattern: '[0-9]+'
+    Description: The 12 digit AWS account number to grant access to.
+    MaxLength: '12'
+    MinLength: '12'
+    Type: String
+    Default: 292230061137
```
Comment (shawndwells) [inline Control-Tower-For-Horizon/s3bucketfiles/horizon_new_account_v4.yaml:46]: Default?

File: Control-Tower-For-Horizon/README.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
+# CrowdStrike with AWS Control Tower 
+
+Setup AWS Control Tower to register new accounts with CrowdStrike Horizon.
+
+## Documentation 
+
+Please see the [Implementation Guide](https://github.com/CrowdStrike/Cloud-AWS/blob/horizon/Control-Tower-For-Horizon/documentation/implementation-guide.md)
```
[the remark this answers is not in the document]
Comment (isimluk) [reply Control-Tower-For-Horizon/README.md:7]: Hold on, our default branch here is not named `main` just yet.

[the remark this answers is not in the document]
Comment (shawndwells) [reply Control-Tower-For-Horizon/README.md:7]: Good catch, ``master`` it is for now

**Pull request merged**